server/src/stateManagement.mjs

```diff
@@ -84,10 +84,23 @@ function addDefaultUser(userId) {
   state[''+userId].isDefaultUserState = true;
 }
 
+// return state based on hierarchy (From lowest priority to highest) global->group->user
 function getState(userId) {
```

Comment:
So we should be returning a combination of the global/group/user state, not just one of them.

Ex: getState('123~abc') should:
* Load the global state
* Load the state for group "~123" if it exists. All fields in this state should be combined with the global state. Any conflicts should have the group state chosen
* Load the state for userId 'abc~123'. All fields in this state should be combined with the newly created group+global state. Any conflicts should have the userId state chosen

This new state is the one that should be returned